Update caching_with_rails.md regarding extension in filename [ci skip]

D-system opened the pull request:
### Motivation / Background

This Pull Request has been created because the documentation is out-of-sync.


### Detail

Since #39164 extension in filename are deprecated (shipped with Rails 6.1)
And has been removed in Rails 7.0.

### Additional information

N/A

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [N/A] Tests are added or updated if you fix a bug or add a feature.
* [N/A] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

p8 commented:
Hi @D-system  👋 

Can the example be changed to use the `:formats` options?
This allows using the html partial in any MIME.
```ruby
render(partial: 'hotels/hotel', collection: @hotels, formats: :html, cached: true)
```

D-system committed:
Doc: update partial rendering with dots in name
Since #39164 extension are deprecated (shipped with Rails 6.1).
And has been removed in Rails 7.0.

D-system commented:
Thanks @p8 for the comment, I updated the PR.

And thanks for the walk to beach during Rails World!
Nice to see you in here.

p8 commented:
Thanks @D-system  😄 
You're welcome and it was nice meeting you.
